### ISSUE TITLE
Add SDK env variable for OTLP TLS skip verify

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-specification#4264]
**What are you trying to achieve?**

I would like to configure OTLP exporter in SDK with env var to skip SSL verification. The variable is not currently supported https://opentelemetry.io/docs/specs/otel/protocol/exporter/ 

What did you expect to see?

**Additional context.**

Created from https://github.com/open-telemetry/opentelemetry-operator/issues/1818

**Comment by marcalff** [comment]
How is this different from `OTEL_EXPORTER_OTLP_INSECURE` ?

**Comment by danielgblanco** [comment]
The difference, as I understand it, is that `OTEL_EXPORTER_OTLP_INSECURE` results in no TLS encryption, while this requested feature would allow for TLS encryption while skipping certification verification.

**Comment by SylvainJuge** [comment]
Hi !

Having a dedicated config option/env variable to disable certificate validation would be great, in particular for use-cases that involve local deployments and self-signed certificates.

Since exporter options are also defined on a per-signal basis, maybe as an alternative we could use a "magic value" with the existing `OTEL_EXPORTER_OTLP_CERTIFICATE` config option and its per-signal alternatives. For example setting `OTEL_EXPORTER_OTLP_CERTIFICATE=*` could work without breaking existing implementations as `*` is not a suitable file path.

**Comment by SylvainJuge** [comment]
After thinking a bit about this, I think there are a few ways we could add this to the [exporter specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) which is already stable.

- option 1: modify the meaning of `Insecure` (`OTEL_EXPORTER_OTLP_INSECURE`) to indicate that when using an `https` endpoint this will disable the server certificate validation.
  - 👍 : does not add a new configuration option, so it's simple
  - 👍 : users should not be relying on this combination, so it should not be a breaking change
  - 👎 : we can't apply this to gRPC as we don't have a protocol part, for gRPC `Insecure` means "do not use TLS", whereas for HTTP we can rely on the `http/https` protocol part.
  - 👎 : modifying the meaning of a stable configuration option

- option 2: modify the meaning of `Certificate File` (`OTEL_EXPORTER_OTLP_CERTIFICATE`) to allow a "magic value" like `*` to disable certificate validation.
  - 👍 : does not add a new configuration option, so it's simple
  - 👍 : as `*` is not a valid file path, users are not currently using this magic value, thus it's not a breaking change.
  - 👎 : modifying the meaning of a stable configuration option, we could introduce this feature as optional and depending on the exporter implementation.

- option 3 : introduce a new `Insecure Skip Verify` (`OTEL_EXPORTER_OTLP_INSECURE_SKIP_VERIFY`) boolean configuration option to disable certificate validation
  - 👍 : no modification of existing stable definitions
  - 👍 : distinct configuration option with explicit definition
  - 👍 : compatible with both HTTPS and gRPC
  - 👎 : introduce a new unstable configuration option in otherwise stable spec (at least short-term).

As I am writing this, the 3rd option seems to be the best option here.

**Comment by SylvainJuge** [comment]
After further investigation, the option 2 is probably the simplest to implement.

I've opened https://github.com/open-telemetry/opentelemetry-specification/pull/5171 to implement the change in spec.

**Comment by jmacd** [comment]
@SylvainJuge Requesting that we discuss this here before opening PRs. I would like to know more about why for "local deployments and self-signed certificates" you can't simply avoid TLS. This looks like a very special and insecure case to be supporting broadly.

I would like to gauge how much interest there is for this feature in the community. Here are our guidelines for relatively small changes. 

https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#smaller-changes

**Comment by SylvainJuge** [comment]
> [@SylvainJuge](https://github.com/SylvainJuge) Requesting that we discuss this here before opening PRs. I would like to know more about why for "local deployments and self-signed certificates" you can't simply avoid TLS. This looks like a very special and insecure case to be supporting broadly.

I agree, for local deployments using plain HTTP might be an option, but there are also the following use-cases that might be relevant to cover:
- help diagnose certificate issues: for example in Java we often get a cryptic error message that does not explicitly says "certificate is not trusted" or which certificate in the chain does not pass verification.
- provide a temporary work-around to deal with cases where the endpoint is using TLS, maybe with a self-signed certificate or when the user can't alter trusted certificates.

Technically, users could already implement something similar by downloading an un-trusted certificate as file and then using `OTEL_EXPORTER_OTLP_CERTIFICATE` to provide a path to it, but this would be portable across all platforms and SDKs without having to deal with openssl or similar command line tools to export the un-trusted certificate.

**Comment by atoulme** [comment]
I support the addition of this env var, and I would like it to be an explicit boolean flag that folks have to set with the term INSECURE as part of it to attract the attention of security teams.

**Comment by SylvainJuge** [comment]
Another potentially good reason to add a new environment variable would be to allow to align with two existing use-cases:
- in collector: `insecure_skip_verify` config option https://github.com/open-telemetry/opentelemetry-collector/blob/a23b4aa293411b5085abf8e6f7c1c703417c92d1/config/configtls/README.md?plain=1#L53
- in OBI with `insecure_skip_verify` config option / `OTEL_EBPF_INSECURE_SKIP_VERIFY` env variable https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/blob/3ba13ca7e6f78aa3526467f676ba095a61b97345/devdocs/config/CONFIG.md?plain=1#L420
